src/fundamental/macro-fundamental.h

```diff
@@ -200,6 +200,19 @@
                 MIN(_c, z);                             \
         })
 
+/* Returns true if the passed integer is a positive power of two */
+#define CONST_ISPOWEROF2(x)                     \
+        ((x) > 0 && ((x) & ((x) - 1)) == 0)
+
+#define ISPOWEROF2(x)                                                  \
+        __builtin_choose_expr(                                         \
```

Comment:
I doubt this does anything worthwhile. Either the compiler does evaluate ISPOWEROF2 at compile time or not. It's more interesting to do this for CONST_ISPOWEROF2 and return void in the non-const case to force compilation error.

Reply:
The only reason I use `__builtin_choose_expr()` here is so that the macro can be used inside constant and non-constant codepaths – but also guarantee side-effect free execution.

i.e. this C expression:

```c
uint8_t x[IS_POWEROF2(2)];
```

should really be a regular array, not a VLA.

Reply:
I guess you have a point. That still allows using `CONST_ISPOWEROF2(x)` which imho, should be a compile-time error like with `ALIGN_TO/CONST_ALIGN_TO`.